docs/protocol.md

```diff
@@ -662,7 +673,233 @@ Response (in case of success)
 ```
 
 </details>
+
+## Sending a heartbeat message to servers
+
+🧭 **RPC Message** > **Query** > **Heartbeat**
+
+A server sends a heartbeat to its peers in order to attest that: 
+
+- it is alive
+- it received a certain set of messages
+
+Heartbeat are to be sent based on channel activity and time intervals.  
+
+RPC 
+
+```json5
+// ../protocol/examples/query/heartbeat/heartbeat.json
+
+{
+    "jsonrpc": "2.0",
+    "id": 5,
+    "method": "heartbeat",
+    "params": {
+        "maps": [
```

Comment:
Why is the name "maps"? Because it contains the messages per channel? The I'd name it something like that, i.e. "messages_by_id"?